### DISCUSSION TITLE
UCUM units: distinguishing between measurement types

### DISCUSSION THREAD

**Discussion by TimVosch** [ucum-org/ucum#371] [General]
Hi group!

I am intriguid by UCUM and have chosen to softly require our the measurement units in our Data Processing System ([SensorBucket](https://github.com/sensorbucket/SensorBucket), forgive some broken urls in the readme, [docs are here](https://sensorbucket.github.io/SensorBucket)) to be UCUM standardized.

Our system is starting to be used by several departments such as groundwater monitoring, air quality and environmental quality. This application brought us to the following question of applicability and distinction.

In a semi hypothetical case, one department measures the groundwatelevel by using tubes fitted with a pressure sensor, while a different department uses pressure sensor to measure the amount of collected rainwater. 
We would like to distinguish between these two measurements. Is this within the scope of UCUM?

The two formats, that distinguish the measurement units, that come to my mind would be:
`mbar[H2O]{groundwater)` and `mbar[H2O]{rainwater}`.

I'd love to hear your thoughts on the subject. 

Sincerely,
Tim van Osch

**Comment by TimVosch** [comment]
@timbrisc Would you happen to know what the/a conventional approach is?

**Comment by timbrisc** [comment]
@TimVosch Hello, Tim. I am not an expert on UCUM syntax but can offer my best explanation plus links to resources which may help further. After reviewing these materials, I can present your specific questions before our committee of experts for their review.

I do not believe the examples you provided are correct. `mbar` is a valid expression of the millibar unit. `[H2O]` is not a unit. Text within curly braces are annotations and not part of the unit per se. If you would like to include these annotations, I believe `mbar{rainwater}` and `mbar{groundwater}` would be valid and sufficient.

I invite you to view the U.S. National Library of Medicine's validator and syntax rules at [https://ucum.nlm.nih.gov/ucum-lhc/demo.html](https://ucum.nlm.nih.gov/ucum-lhc/demo.html).

We certainly welcome and appreciate the use of UCUM in your project. Please keep us informed of your activities.

**Reply by TimVosch** [reply]
Hi Tim, Thank you for sharing your knowledge. I see that the validator also identifies the unit as invalid. 
Thank you for proposing this question to the committee of experts. I look forward to their evaluation and feedback.

Sincerely,
Tim van Osch

**Comment by chgessner** [comment]
Right, [H2O] is not a unit. Note however, that there is a UCUM unit "m[H2O]" which stands for "meter of water column". This is in fact another way to express pressure quantities. Therefore m[H2O] is convertible to Pascal, and hence convertible to mbar. You can even add a prefix, yielding UCUM codes like "cm[H2O]" or "mm[H2O]".

**Reply by TimVosch** [reply]
Right, I've read the spec a bit further in depth and see the example on square brackets. Being that square brackets change the meaning of the atom unit. Just like you say meter is a unit of distance, but m[H2O] is a unit of pressure.

Is the conversion of m[H2O] set in the spec? Or could it technically be a different fluid (say mercury)?

Back to the use case I've given, it is probably best to go with mbar{groundwater} and mbar{rainwater}, as that is the unit the sensor measures at and thus the "purest" unit (i.e. the least conversion errors). After that, a conversion to m[H2O]{groundwater} is still possible.

**Reply by chgessner** [reply]
Besides "m[H2O] you'll find "m[Hg]" as a UCUM atom. Not for other substances. Conversions to SI units are also somewhere in the XML reference and in the HTML specs.
